src/coreclr/inc/corinfo.h

```diff
@@ -2277,8 +2277,7 @@ class ICorStaticInfo
     //    bufferSize - Pointer to buffer length. Must not be nullptr
     //
     // Return Value:
-    //    Full length of the source UTF8 representation so can be bigger than bufferSize.
-    //    Includes null-terminator.
+    //    Bytes written to the given buffer, the range is [0..bufferSize]
```

Comment:
I'd like the JIT to be able to print the exact class name even for large lengths, so this means `printClassName` is going to be inconsistent with this API design.

I guess we can use a doubling strategy, but it seems unnecessary to me.

Reply:
I would add an extra `size* pRequiredBufferSize = nullptr` argument for this. It allows you to get the full buffer size only if you actually want it.

Reply:
Do you want me to add an additional argument here just for consistency or @jakobbotsch you plan to rely on it? Extending JIT-EE is quite painful 😞

Reply:
I would add it so that we have full flexibility and do not have to deal with this ever again. @jakobbotsch ?

Reply:
Sounds fine to me.

Nit: the returned value is going to be in range `[0..bufferSize)` if it does not include null terminator.

Reply:
It does include it so should be `bufferSize]`

Reply:
I was under the impression we had converged on not including the null terminator to be in line with `sprintf_s` family of functions?

Reply:
I am not sure I understand the final decision either - a bit lost in the comments, if we don't want to null-terminate it then we basically are back to the shape that was before we start changing it back and forth 😄 except the `size* pRequiredBufferSize` parameter

Reply:
The API is null terminating always, just the return value does not include the null terminator. Then it is consistent with C style string functions.

Reply:
Or do we want to still null-terminate it there but not include the `\0` into the return value? Ok

Reply:
Ah, I see